### PR TITLE
Camera calibration for Kinect and Realsense

### PR DESCRIPTION
**Pull request by TheoMadaBoy**
### Addition of the Calibration Package:
- Added launch files that execute all steps required for both camera calibrations (Kinect : Eye-on-base, Realsense : Eye-in-hand)

### Addition of the ros2_markertracker Package:
- Implements functionality to detect Aruco markers and publish their transformations (TF).
- For more details, refer to the original repository: [ros2_markertracker](https://github.com/Veilkrand/ros2_markertracker/tree/master).

### Enhancements in the calibration_camera Package:
- Added a node to convert a PoseArray to TF transformations.
- Included configuration .yaml files for the intrinsic parameters of the cameras, which are necessary for calibration.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,80 @@
...
+rqt
+```
+Go to -> Plugins -> Services -> Service caller
+
+Find, in the dropdown menu, the service called : "/dsr01/system/set_robot_mode" and set the value to 0, 
+
+Command to launch the calibration for the Azure Kinect : 
+
+```bash
+ros2 run camera_calibration kinect_eye_on_base_calibration.launch.py
```
Comment (TheoMadaBoy) [inline README.md]: launch

File: README.md [hunk 2]
```diff
@@ -1,2 +1,80 @@
...
+```
+
+Command to launch the calibration for the Real Sense : 
+```bash 
+ros2 launch camera_calibration rs_eye_on_hand_calibrate.launch.py
+```
+
+In another terminal, you will need to make a service call to calibrate the camera with : 
+```bash
+ros2 service call /hand_eye_calibration/capture_point std_srvs/srv/Trigger {}
```
Comment (will-44) [inline README.md]: cette commande declanche une erreur pour la kinect

File: camera_calibration/config/kinect.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+%YAML:1.0
```
Comment (will-44) [inline camera_calibration/config/kinect.yaml:1]: a terme ce serait bien de juste prendre les info de camera info pour plus de versatilité

File: README.md [hunk 4]
```diff
@@ -1,2 +1,80 @@
...
+Command to launch the calibration for the Azure Kinect : 
+
+```bash
+ros2 run camera_calibration kinect_eye_on_base_calibration.launch.py
+```
+**Important note** : The kinect has a fish-eye lense. In the calibration file, there is already a function
+in place that rectifies the image. There is also a bash command that has the same effect. It needs to be run 
+when you have started the azure kinect launch file. It is :
+```bash
+ros2 run image_proc rectify_node --ros-args -r __node:=rectify_rgb --remap image:=rgb/image_raw --remap image_rect:=rgb/image_rect_raw
```
Comment (will-44) [inline README.md]: ce node est deja executé dans le launch file de la kinect

File: camera_calibration/launch/kinect_eye_on_base_calibration.launch.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    #Nouveau nœud pour la détection des marqueurs fiduciaires
+    markertracker_node = Node(
+        package='ros2_markertracker',
+        executable='markertracker_node',
+        # output="screen",
+        namespace="/ros2_markertracker",
+        parameters=[{
+            "input_image_topic": "/rgb/image_rect_raw",
+            "publish_topic_image_result": True,
+            "path_to_camera_file": '/home/coro/ros2_ws/src/tool_box/camera_calibration/config/kinect.yaml',
```
Comment (TheoMadaBoy) [inline camera_calibration/launch/kinect_eye_on_base_calibration.launch.py]: path relatif et pas absolut

**Review by will-44 — APPROVED** [review]

**Pull request merged**